static/src/javascripts/projects/common/modules/commercial/track-ad.js

```diff
@@ -0,0 +1,30 @@
+define([
+    'Promise',
+    'common/utils/mediator'
+], function(Promise, mediator) {
+    return trackAd;
+
+    function trackAd(id) {
+        return new Promise(function (resolve, reject) {
+            var onAdLoaded = function (event) {
+                if (event.slot.getSlotElementId() === id) {
+                    unlisten();
+                    resolve(!event.isEmpty);
```

Comment:
Should _trackAd_'s interface be to return a success code, or should the promise return the event for the advert in question, given its generic-sounding name?